tools/odbc/result.cpp

```diff
@@ -33,20 +62,151 @@ SQLRETURN SQLGetData(SQLHSTMT statement_handle, SQLUSMALLINT col_or_param_num, S
 		}
 
 		switch (target_type) {
-		case SQL_C_SLONG:
-			D_ASSERT(((size_t)buffer_length) >= sizeof(int));
-			Store<int>(val.GetValue<int>(), (duckdb::data_ptr_t)target_value_ptr);
+		case SQL_C_SSHORT: {
+			D_ASSERT(((size_t)buffer_length) >= sizeof(SQLSMALLINT));
```

Comment:
This code snippet seems repeated several types here with only the template parameters changing, could you extract this into a function, e.g.:

```cpp
template<class T> 
SQLRETURN GetInternalValue(duckdb::OdbcHandleStmt *stmt, duckdb::Value &val, SQLLEN buffer_length, SQLPOINTER target_value_ptr) {
    D_ASSERT(((size_t)buffer_length) >= sizeof(T));
    try {
        Store<T>(val.GetValue<T>(), (duckdb::data_ptr_t)target_value_ptr);
        return SQL_SUCCESS;
    } catch (std::exception &ex) {
        stmt->error_messages.emplace_back(ex.what());
        return SQL_ERROR;
    }
}
```

And then call it like so:

```cpp
return GetInternalValue<SQLSMALLINT>(stmt, val, buffer_length, target_value_ptr);
```